test_rmw_implementation/test/test_serialize_deserialize.cpp

```diff
@@ -112,6 +119,66 @@ TEST_F(CLASSNAME(TestSerializeDeserialize, RMW_IMPLEMENTATION), clean_round_trip
     rmw_get_error_string().str;
 }
 
+TEST_F(CLASSNAME(TestSerializeDeserialize, RMW_IMPLEMENTATION), clean_round_trip_for_c_bounded_message) {
+  const rosidl_message_type_support_t * ts{
+    ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, BoundedSequencesNoStrings)};
+  test_msgs__msg__BoundedSequencesNoStrings input_message{};
+  test_msgs__msg__BoundedSequencesNoStrings output_message{};
+  ASSERT_TRUE(test_msgs__msg__BoundedSequencesNoStrings__init(&input_message));
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    test_msgs__msg__BoundedSequencesNoStrings__fini(&input_message);
+  });
+  ASSERT_TRUE(test_msgs__msg__BoundedSequencesNoStrings__init(&output_message));
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    test_msgs__msg__BoundedSequencesNoStrings__fini(&output_message);
+  });
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  rmw_serialized_message_t serialized_message = rmw_get_zero_initialized_serialized_message();
+  ASSERT_EQ(
+    RMW_RET_OK, rmw_serialized_message_init(
+      &serialized_message, 0lu, &default_allocator)) << rmw_get_error_string().str;
+
```

Comment:
@MiguelCompany an `OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT` block finalizing `serialized_message` is missing here.